src/operations/extension/reduction_partial_rows.hpp

```diff
@@ -44,7 +44,7 @@ class ReductionPartialRows {
   using params_t = ReductionRows_Params<index_t, element_t, ClSize, WgSize>;
 
   /* Neutral value for this reduction operator */
-  static constexpr value_t init_val = operator_t::template init<output_t>();
+  static const value_t init_val;
```

Comment:
I'm not following why you did this

Reply:
This worries me a little bit - whether it has any impact on performance - but the reason is that `cl::sycl::half` is not `constexpr`.

Reply:
This can affect the performance on other types.

Reply:
I can take another look, but I'm not sure how best to proceed. I guess I should measure and if it doesn't have an impact it can go in as-is?

Reply:
@ProGTX Have you had a chance to check the performance impact of this change?

Reply:
It is very important that the init value to be known at compile time. In this case, the compiler will be able to flatten the value, and later on when it is used in a kernel or for loop, it can unroll the for loop and promote the array element to a register. Can you please revert that change?

Reply:
The problem is with the lack of `constexpr`, so this cannot be reverted for now. @mcleary is getting performance numbers, but so far it doesn't seem like there's a performance regression.

Reply:
Yes, I understand, Since the issue will be fixed in SYCL-2020 (Hopefully), 
Can add a comment explaining the issue followed by TODO/FIXME saying the issue will be fixed  (comment out the Constexpr line heare) once SYCL-2020 is feature is added to computeCpp.

Reply:
Also, the `int_val` variable should have an initial value  `operator_t::template init<output_t>();` and the initial value is not necessarily zero. Where do you set that?

Reply:
I've added TODOs: https://github.com/codeplaysoftware/sycl-blas/pull/251/commits/3cfa6b09ac2b423e048f8fe8a6ea54ec1b79f381

The reason for this particular line, why it's not initialized at this point, is because of C++ rules on initializing static variables. In this case it can't be initialized inline anymore because it's not a constant initialization anymore, so it had to be moved out of the class.

Reply:
Thanks! As long as it is initialized somewhere before using, it is ok. It will be fixed once we have the `constexpr` for half.